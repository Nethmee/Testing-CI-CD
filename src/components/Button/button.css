.sampleButton{
 
    background-color:#3EAF86;
    border-radius: 50%;
    border-style: solid;
    border-width: 1px;
    background-color:"#3EAF86";
    margin-top: 40vh;
    width:10vw;
    height:10vw; 
    color:#fff;
    font-size: 22px;
    font-weight: bold;
    box-shadow: 2px 2px 3px #999;
    text-align: center;
    display:flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    
}

