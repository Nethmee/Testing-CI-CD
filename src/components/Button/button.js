import React from "react";
import "./button.css";

export default function button({label}) {
  return (
    <div className="sample">
      
      <div data-testid="button" className="sampleButton"> {label} </div>
    </div>
  );
}
button.defaultProps = {
    label:'this is a button'
}