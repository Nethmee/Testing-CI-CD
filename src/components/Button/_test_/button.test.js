import React from 'react';
import ReactDOM from 'react-dom';
import  Button from './../button';
import  {isTsAnyKeyword} from '@babel/types';
import {render,cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import renderer from "react-test-renderer"

afterEach(cleanup)
//checking if the component renders without crashing

it('renders without crashing', ()=>{
    const div = document.createElement('div');
    ReactDOM.render(<Button></Button>,div)
})


it('renders the button correctly',()=>{

    const {getByTestId} = render(<Button label={'click this test'}></Button>)
    expect(getByTestId('button')).toHaveTextContent('click this test')
})

it('renders the button correctly',()=>{

    const {getByTestId} = render(<Button label={'Save'}></Button>)
    expect(getByTestId('button')).toHaveTextContent('Save')
})


//snapshot testing
it("matches snapshot",()=>{
  const tree =  renderer.create(<Button label="save me"></Button>).toJSON();
  expect(tree).toMatchSnapshot();
}
)

it("it matches the snapshot 2", ()=>{

    //they have a snapshot 
    const tree = renderer.create(<Button></Button>).toJSON();
    expect(tree).toMatchSnapshot()
})

 
