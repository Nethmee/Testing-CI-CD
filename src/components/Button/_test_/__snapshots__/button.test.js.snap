// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches the snapshot 2 1`] = `
<div
  className="sample"
>
  <div
    className="sampleButton"
    data-testid="button"
  >
     
    this is a button
     
  </div>
</div>
`;

exports[`matches snapshot 1`] = `
<div
  className="sample"
>
  <div
    className="sampleButton"
    data-testid="button"
  >
     
    save me
     
  </div>
</div>
`;
